src/JsonSchema/SchemaFactory.php

```diff
@@ -113,33 +127,66 @@ public function buildSchema(string $className, string $format = 'json', string $
+            if (false === ($serializerContext[AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES] ?? true)) {
+                $definition['additionalProperties'] = false;
+            }
+
+            // see https://github.com/json-schema-org/json-schema-spec/pull/737
+            if (
+                Schema::VERSION_SWAGGER !== $version &&
+                null !== $resourceMetadata &&
+                (
+                    (null !== $operationType && null !== $operationName && null !== $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'deprecation_reason', null, true)) ||
+                    null !== $resourceMetadata->getAttribute('deprecation_reason', null)
+                )
+            ) {
+                $definition['deprecated'] = true;
+            }
+            // externalDocs is an OpenAPI specific extension, but JSON Schema allows additional keys, so we always add it
+            // See https://json-schema.org/latest/json-schema-core.html#rfc.section.6.4
+            if (null !== $resourceMetadata && null !== $iri = $resourceMetadata->getIri()) {
+                $definition['externalDocs'] = ['url' => $iri];
+            }
+        } else { // New interface
+
+            if (null !== $resourceMetadata && null !== $description = $resourceMetadata->getOperation($operationName)->description) {
+                $definition['description'] = $description;
+            }
+
+            // additionalProperties are allowed by default, so it does not need to be set explicitly, unless allow_extra_attributes is false
+            // See https://json-schema.org/understanding-json-schema/reference/object.html#properties
+            if (false === ($serializerContext[AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES] ?? true)) {
+                $definition['additionalProperties'] = false;
+            }
+
+            // see https://github.com/json-schema-org/json-schema-spec/pull/737
+            if (
+                Schema::VERSION_SWAGGER !== $version &&
+                null !== $resourceMetadata &&
+                (
+                    (null !== $operationType && null !== $operationName && null !== $resourceMetadata->getOperation($operationName)->deprecationReason) ||
+                    null !== $resourceMetadata->getOperation($operationName)->deprecationReason
+                )
+            ) {
+                $definition['deprecated'] = true;
+            }
+            // externalDocs is an OpenAPI specific extension, but JSON Schema allows additional keys, so we always add it
+            // See https://json-schema.org/latest/json-schema-core.html#rfc.section.6.4
+            // TODO: Surement pas bon ici le ->uriTemplate
+            if (null !== $resourceMetadata && null !== $iri = $resourceMetadata->getOperation($operationName)->uriTemplate) {
+                $definition['externalDocs'] = ['url' => $iri];
+            }
```

Comment:
Je pense que c'est pas bon ca, essaie plutot de faire un truc comme ca:
- identifie les propriétés qui proviennent des metadonnéés (description, externalDocs) 
- créé une fonciton
- on appelle tout le temps `null !== $resourceMetadata` ca sert a rien, autant faire qqch du genre:

if ($resourceMetadata) {
  $definition = $resourceMetadata instanceof ResourceMetadata ? $this->expandDefinitionFromMetadata($resourceMetadata, $operationName) : $this->expandDefinitionFromResourceCollection($resourceMetadata, $operationName);
}
```
- iri => Types pas uriTemplate